src/dtemplate.d

```diff
@@ -7187,6 +7187,14 @@ public:
             else if (isParameter(o))
             {
             }
+            else if (Tuple tup = isTuple(o))
+            {
+                // Expand tuple and insert its elements into tiargs
+                tiargs.remove(j);
+                tiargs.insert(j, &tup.objects);
+                j--;
+                continue;
+            }
```

Comment:
I think the problem is that a `Tuple` object appears on template argument list.
As far as I know, `Tuple` is created during template instantiation process, but after its done, the variadic object list will be closed into `TupleDeclaration` always, to be accessible as a named symbol. Therefore, if a `Tuple` object is returned as a result of preceding semantic analysis, it's a bug.

Anyway, we need to get a minimized test case to reproduce the issue.